Tests/appstoreconnect-cliTests/Operations/ReadPreReleaseVersionOperationTests.swift

```diff
@@ -0,0 +1,196 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+@testable import AppStoreConnectCLI
+import AppStoreConnect_Swift_SDK
+import Combine
+import XCTest
+
+final class ReadPreReleaseVersionOperationTests: XCTestCase {
+    typealias Operation = ReadPreReleaseVersionOperation
+    typealias Options = Operation.Options
+    typealias OperationError = ReadPreReleaseVersionOperation.Error
+
+    let successResponseRequestor = OneEndpointTestRequestor(response: { _ in
+        Future({ $0(.success(onePreRealeseVersionResponse)) }) }
+    )
+
+    let noResponseRequestor = OneEndpointTestRequestor(response: { _ in
+        Future{ $0(.success(noPreReleaseVersionResponse)) }}
+    )
+
+    let notUniqueRequestor = OneEndpointTestRequestor(response: { _ in
+        Future{ $0(.success(notUniqueResponse)) }}
+    )
+
+    func testOnePreReleaseVersion() throws {
+        let operation = Operation(options: Options(filterAppId: "1504341572", filterVersion: "1.0"))
+
+        let result = Result {
+            try operation.execute(with: successResponseRequestor).await()
+        }
+
+        switch result {
+        case .success(let output):
+            XCTAssertEqual(output.preReleaseVersion.attributes?.version, "1.0")
+        default:
+            XCTFail("Error in getting prereleaseVersion response")
+        }
+    }
+
+    func testNoPreReleaseVersion() {
+        let operation = Operation(options: Options(filterAppId: "1504341572", filterVersion: "0.0"))
+        let expectedError = OperationError.noVersionExists
+
+        let result = Result {
+            try operation.execute(with: noResponseRequestor).await()
+        }
+
+        switch result {
+        case .failure(let error as OperationError):
+            XCTAssertEqual(error.errorDescription, expectedError.errorDescription)
+        default:
+            XCTFail("Expected failure with \(expectedError) but got: \(result)")
+        }
+    }
+
+    func testNotUniquePreReleaseVersion() {
+        let operation = Operation(options: Options(filterAppId: "1504341572", filterVersion: "1.0"))
+        let expectedError = OperationError.versionNotUnique
+
+        let result = Result {
+            try operation.execute(with: notUniqueRequestor).await()
+        }
+
+        switch result {
+        case .failure(let error as OperationError):
+            XCTAssertEqual(error.errorDescription, expectedError.errorDescription)
+        default:
+            XCTFail("Expected failure with \(expectedError) but got: \(result)")
+        }
```

Comment:
```suggestion
        let operation = Operation(options: Options(filterAppId: "1504341572", filterVersion: "1.0"))
        let expectedError = OperationError.versionNotUnique

       XCTAssertThrows(try operation.execute(with: notUniqueRequestor).await()) { error in 
           XCTAssertEqual(error, OperationError.versionNotUnique)
        }
```

Assuming OperationError implements `Equatable`.

Reply:
I had to cast error as! OperationError